@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -58,7 +58,7 @@ type Literal = "boolean" | "number" | "string";
 /**
  * @hidden
  */
-type ExprKind = "var" | "in" | Literal | UnaryOp | RelationalOp | EqualityOp | LogicalOp;
+type ExprKind = "var" | "in" | "length" | Literal | UnaryOp | RelationalOp | EqualityOp | LogicalOp;
```

Comment:
What happens if there is an attribute "length"?

Reply:
It will still work, please note double brackets '(...)', and function call semantics. I have checked it by changing 'length' function name to 'ref', so we could even use such conditions:
"when": "$layer == 'roads' && ... && has(ref) && (ref(ref) == 3)"

Reply:
okay, but I think readability of this is bad, but hopefully this will never happen.
For the length function I would prefer to rename it to sth. like strLength b/c the probability that there will be a length attribute in the data is quite high (i.e. we have it already for truck restrictions).